### PR TITLE
fix: disambiguate users and guest authors in Author List exclusions

### LINKED ISSUE #1150: Author List – guest authors disambiguation
Similarly to #1134 (fixed in #1143), the Author List block should also handle the issue of CAP & WP authors having the same ID.

## To reproduce

1. On a site with a CAP author having the same ID as a WP user insert an Author List block
2. Exclude one of these users
3. Observe both are selected (but only one appears in "X author selected" message):

image

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

#1143 fixed a bug in the Author Profile block when authors and guest authors had the same ID. This PR fixes a similar bug in the exclusions attribute of the Author List block. However, it also requires https://github.com/Automattic/newspack-plugin/pull/1679 to be released to the `newspack-components` NPM package, so consider this a draft PR until that release is deployed.

Note that this fix won't fix pre-existing instances of the block, although it does handle the flat array structure of the attribute for backwards compatibility. Any blocks affected by the bug will need to be rebuilt.

Closes #1150.

### How to test the changes in this Pull Request:

1. In a local environment, checkout https://github.com/Automattic/newspack-plugin/pull/1679 and this PR.
2. Confirm that #1150 is not replicable: only the exact user or guest author you select should be checked in the block controls, and only that user or guest author should be excluded in the editor preview and on the front-end.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by adekbadek — CHANGES_REQUESTED** [review]
This does not resolve issue #1150 for me, it still can be reproduced.

**Comment by dkoo** [comment]
@adekbadek I remember now why this was blocked—it requires https://github.com/Automattic/newspack-plugin/pull/1679, which was recently released into the Newspack Components NPM package as v2.1.0. That package is updated in #1188, so once that's merged the fix here should start working. I'll update the status here to "On hold" until #1188 is merged.

File: src/blocks/author-list/edit.js [hunk 1]
```diff
@@ -245,6 +248,8 @@ const AuthorList = ( { attributes, clientId, setAttributes } ) => {
 								const total = parseInt( response.headers.get( 'x-wp-total' ) || 0 );
 								const suggestions = await response.json();
 
+								console.log( suggestions );
```
[the remark this answers is not in the document]
Comment (dkoo) [reply src/blocks/author-list/edit.js]: Whoops! 7c3374e

**Comment by dkoo** [comment]
`newspack-components` package has been updated and released; re-opening this PR.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**